sound/soc/intel/boards/sof_da7219_max98373.c

```diff
@@ -383,7 +401,8 @@ static int audio_probe(struct platform_device *pdev)
 			.no_pcm = 1,
 			.dpcm_playback = 1,
 			.ignore_pmdown_time = 1,
-			SND_SOC_DAILINK_REG(ssp1_pin, dummy, platform) };
+			.init = speaker_amp_init,
```

Comment:
you could use a different routine so you don't do the test multiple times and separate the two cleanly?

Reply:
Ack, I also debate which way is better here, will update in next version.